apps/staking/app/register/[nodeId]/Registration.tsx

```diff
@@ -0,0 +1,572 @@
+    mode: 'onChange',
+  });
+
+  const formMulti = useForm<MultiRegistrationFormSchema>({
+    resolver: zodResolver(formMultiSchema),
+    defaultValues: {
+      rewardsAddress: queryParamFields?.rewardsAddress ?? address,
+      stakeAmount:
+        queryParamFields?.stakeAmount ??
+        bigIntToString(
+          bigIntMin(SESSION_NODE_MIN_STAKE_MULTI_OPERATOR, balanceValue ?? 0n),
+          SENT_DECIMALS,
+          decimalDelimiter
+        ),
+      autoActivate: queryParamFields?.autoActivate ?? true,
+      operatorFee: queryParamFields?.operatorFee ?? '',
+      reservedContributors: queryParamFields?.reservedContributors ?? [],
+    },
+    reValidateMode: 'onChange',
+    mode: 'onChange',
+  });
+
+  return (
+    <RegistrationContext.Provider
+      value={{
+        address,
+        changeTab,
+        contract,
+        dict,
+        formMulti,
+        formSolo,
+        handleTabBackButtonClick,
+        isError,
+        isSubmitting,
+        isSuccess,
+        mode,
+        nodeType,
+        props: {
+          blsKey,
+          ed25519PubKey,
+          blsSignature,
+          ed25519Signature,
+          preparedAt,
+        },
+        setBackButtonClickCallback,
+        setContract,
+        setIsError,
+        setIsSubmitting,
+        setIsSuccess,
+        setMode,
+        setStartData,
+        tab,
+        tabHistory,
+        getQueryParams,
+        pushQueryParam,
+      }}
+    >
+      {children}
+    </RegistrationContext.Provider>
+  );
+}
+
+export function useRegistrationWizard() {
+  const context = useContext(RegistrationContext);
+
+  if (context === undefined) {
+    throw new Error('useRegistrationWizard must be used within RegistrationProvider');
+  }
+
+  return context;
+}
+
+export function useTabDictionary(tab: REG_TAB) {
+  const tabName = parseTab(tab);
+  return useTranslations(`actionModules.registration.${tabName}`);
+}
+
+function getTab(tab: REG_TAB) {
+  switch (tab) {
+    case REG_TAB.START:
+      return <StartTab />;
+    // SOLO
+    case REG_TAB.REWARDS_ADDRESS_INPUT_SOLO:
+      return <RewardsAddressInputSoloTab />;
+    case REG_TAB.SUBMIT_SOLO:
+      return <SubmitSoloTab />;
+    case REG_TAB.SUCCESS_SOLO:
+      return <SuccessSoloTab />;
+    // MULTI
+    case REG_TAB.STAKE_AMOUNT:
+      return <StakeAmountTab />;
+    case REG_TAB.OPERATOR_FEE:
+      return <OperatorFeeTab />;
+    case REG_TAB.REWARDS_ADDRESS:
+      return <RewardsAddressTab />;
+    case REG_TAB.REWARDS_ADDRESS_INPUT_MULTI:
+      return <RewardsAddressInputMultiTab />;
+    // TODO: Implement the reserve slots
+    // case REG_TAB.RESERVE_SLOTS:
+    //   return <ReserveSlotsTab />;
+    // case REG_TAB.RESERVE_SLOTS_INPUT:
+    //   return <ReserveSlotsInputTab />;
+    case REG_TAB.AUTO_ACTIVATE:
+      return <AutoActivateTab />;
+    case REG_TAB.SUBMIT_MULTI:
+      return <SubmitMultiTab />;
+    case REG_TAB.SUCCESS_MULTI:
+      return <SuccessMultiTab />;
+
```

Comment:
If you are going to have a new line break then I would like it above all the section comments please? Otherwise remove this one 🙏